src/routes/console/project-[project]/messaging/+layout.ts

```diff
@@ -2,9 +2,34 @@ import Breadcrumbs from './breadcrumbs.svelte';
 import Header from './header.svelte';
 import type { LayoutLoad } from './$types';
 
-export const load: LayoutLoad = async () => {
+import { queries, queryParamToMap } from '$lib/components/filters';
+import { PAGE_LIMIT } from '$lib/constants';
+import { getLimit, getPage, getQuery, getSearch, pageToOffset } from '$lib/helpers/load';
+import { sdk } from '$lib/stores/sdk';
+import { Query } from '@appwrite.io/console';
+
+import type { Load } from '@sveltejs/kit';
+
+// I **think** this should be enough, tbh
+const loadProviders = async ({ url, route }: Parameters<Load>[0]) => {
+    const page = getPage(url);
+    const limit = getLimit(url, route, PAGE_LIMIT);
+    const offset = pageToOffset(page, limit);
+    const query = getQuery(url);
+    const parsedQueries = queryParamToMap(query || '[]');
+    queries.set(parsedQueries);
+    const search = getSearch(url);
+
+    return sdk.forProject.messaging.listProviders(
+        [Query.limit(10), Query.offset(offset), Query.orderDesc(''), ...parsedQueries.values()],
+        search || undefined
+    );
```

Comment:
I'm not sure if we should do it this way because this is for loading the providers for the providers page where all the filtering would make sense. 

I think it'd be cleaner to make an API call on the mount of the wizard and filter for the specific provider type and limit of 1 because we only need to check if at least 1 matching provider exists.

Reply:
you mean on step1.svelte's onMount?